helpers/useSession.tsx

```diff
@@ -1,23 +1,37 @@
 import useSWR from 'swr'
 import { User } from './models/User'
+import _ from 'lodash'
 
 const SERVER_URL = process.env.SERVER_URL
 
 export const fetcher = (url: string) =>
-  fetch(url, { credentials: 'include' }).then(r => r.json())
+  fetch(url, {
+    method: 'POST',
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      query: `
+            query {
+              session {
+                user {
+                    username
+                }
+              }
+            }
+          `
+    })
+  }).then(r => r.json())
 
 export type SessionData = {
-  data: {
-    userInfo?: User
-    errorMessage?: string
-    success: boolean
-  }
-  error?: boolean
+  session: User | null
 }
 
 const useSession = (): SessionData => {
-  const { data, error } = useSWR(`${SERVER_URL}/session`, fetcher)
-  return { data, error }
+  const { data } = useSWR(`${SERVER_URL}`, fetcher)
```

Comment:
We should just be replacing `useSWR` with the GET_APP query right?

Reply:
Unless you only want to query for the username, you can create a new query string for that

Reply:
I get an error If I call useQuery() within useSession

Reply:
In that case I would probably replace `useSession` everywhere with the `useQuery` hook. Makes more sense :) Tbh, when I added `useSession` it wasn't very useful in the beginning anyways lol

Reply:
@coltonehrman this was tried, but failed. I think that could be a step 2 migration, otherwise this PR may never finish.

Reply:
STEP 2 migration should take out context since we don't need it. This can be resolved.